src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +29,33 @@
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              Todo #{chosenTodo.id}
```

Comment:
You are generating key on render. This can lead to performance issues and unexpected behavior in your application. Check out this [article](https://medium.com/blackrock-engineering/5-common-mistakes-with-keys-in-react-b86e82020052) to understand why it's a bad practice.